lib/ruby_lsp/requests/support/syntax_tree_formatting_runner.rb

```diff
@@ -0,0 +1,39 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "syntax_tree/cli"
+require "singleton"
+
+module RubyLsp
+  module Requests
+    module Support
+      # :nodoc:
+      class SyntaxTreeFormattingRunner
+        extend T::Sig
+        include Singleton
+
+        sig { void }
+        def initialize
+          @options =
+            T.let(
+              begin
+                opts = SyntaxTree::CLI::Options.new
+                opts.parse(SyntaxTree::CLI::ConfigFile.new.arguments)
+                opts
+              end,
+              SyntaxTree::CLI::Options,
+            )
+        end
+
+        sig { params(_uri: String, document: Document).returns(T.nilable(String)) }
+        def run(_uri, document)
```

Comment:
Can we remove `uri` here? It's not used and these runners don't necessarily need to comply to the same exact interface.

Reply:
Sure thing!

I called that out in the first commit. The thinking was that establishing an interface would be handy for other formatters, but I'm game for keeping this more focused the bits this new runner needs at this point.

Reply:
Oh, I didn't see the commit descriptions. Apologies!

That's a fair point. I guess it could be worth having an interface and it doesn't hurt to keep it.